src/main/java/org/opensearch/flowframework/workflow/ModelGroupStep.java

```diff
@@ -0,0 +1,141 @@
+
+        CompletableFuture<WorkflowData> registerModelGroupFuture = new CompletableFuture<>();
+
+        ActionListener<MLRegisterModelGroupResponse> actionListener = new ActionListener<>() {
+            @Override
+            public void onResponse(MLRegisterModelGroupResponse mlRegisterModelGroupResponse) {
+                logger.info("Model group registration successful");
+                registerModelGroupFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(
+                            Map.entry("model_group_id", mlRegisterModelGroupResponse.getModelGroupId()),
+                            Map.entry("model_group_status", mlRegisterModelGroupResponse.getStatus())
+                        )
+                    )
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to register model group");
+                registerModelGroupFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+            }
+        };
+
+        String modelGroupName = null;
+        String description = null;
+        List<String> backendRoles = new ArrayList<>();
+        AccessMode modelAccessMode = null;
+        Boolean isAddAllBackendRoles = false;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        modelGroupName = (String) content.get(NAME_FIELD);
+                        break;
+                    case DESCRIPTION:
+                        description = (String) content.get(DESCRIPTION);
+                        break;
+                    case BACKEND_ROLES_FIELD:
+                        backendRoles = (List<String>) content.get(BACKEND_ROLES_FIELD);
+                    case MODEL_ACCESS_MODE:
+                        modelAccessMode = (AccessMode) content.get(MODEL_ACCESS_MODE);
+                    case ADD_ALL_BACKEND_ROLES:
+                        isAddAllBackendRoles = (Boolean) content.get(ADD_ALL_BACKEND_ROLES);
+                    default:
+                        break;
+                }
+            }
+        }
+
+        if (modelGroupName == null) {
+            registerModelGroupFuture.completeExceptionally(
+                new FlowFrameworkException("Model group name is not provided", RestStatus.BAD_REQUEST)
+            );
+        } else {
+            MLRegisterModelGroupInputBuilder builder = MLRegisterModelGroupInput.builder();
+            builder.name(modelGroupName);
+            if (description != null) {
+                builder.description(description);
+            }
+            if (backendRoles != null && backendRoles.size() > 0) {
```

Comment:
You're defining this as an empty list to start with, so it should never be null. And `isEmpty()` is a few nanoseconds faster than getting the size.
```suggestion
            if (!backendRoles.isEmpty()) {
```